src/proposals/Proposals.jl

```diff
@@ -245,5 +246,154 @@ function (prop::RStagger)(rng::AbstractRNG,
     return u, v, logl, ncall
 end
       
+"""
+    Proposals.Slice(;slices=5, scale=1)
+
+Propose a new live point by a series of random slices away from an existing live point.
+This is a standard _Gibbs-like_ implementation where a single multivariate slice is a combination of `slices` univariate slices through each axis.
+
+## Parameters
+- `slices` is the minimum number of slices
+- `scale` is the proposal distribution scale, which will update _between_ proposals.
+"""
+@with_kw mutable struct Slice <: AbstractProposal
+    slices = 5
+    scale = 1
+    # fmove = 0.9 this parameter to be included in Proposals.HSlice
+    # max_move = 100 ?? not sure if this parameter is to be included only in Proposals.HSlice or in all 3 Slice Proposals
+    
+    @assert slices ≥ 1 "Number of slices must be greater than or equal to 1"
+    @assert scale ≥ 0 "Proposal scale must be non-negative"
+end
 
+function (prop::Slice)(rng::AbstractRNG,
+                       point::AbstractVector,
+                       logl_star,
+                       bounds::AbstractBoundingSpace,
+                       loglike,
+                       prior_transform;
+                       kwargs...)
+    # setup
+    n = length(point)
+    slices_init = prop.slices
+    nc = nexpand = ncontract = 0
+    fscale = [] 
+    axlens = []
+    local idxs, r, u, u_prop, v_prop, logl_prop, logl_l, logl_r
+    
+    # modifying axes and computing lengths
+    axes = prop.scale * transpose(axes)  # scale based on past tuning
+    for axis in axes
+        append!(axlens, axis)
+    end
+    
+    # slice sampling loop
+    for it in 1:slices
+        
+        # shuffle axis update order
+        idxs = shuffle!(rng, Vector(1:n))
+        
+        # slice sample along a random direction
+        for idx in idxs
+            
+            # select axis
+            axis = axes[idx]
+            axlen = axlens[idx]
+            
+            # define starting window
+            r = rand()  # initial scale/offset
+            u_l =  u - r * axis  # left bound
+            if all(u -> -0.5 < u < 1.5, u_l)    ## ??check if this non-periodic boundary condition is correct, it allows to exceed the unit cube 
+                v_l = prior_transform(u_l)
+                logl_l = loglike(v_l)
+            else
+                logl_l = -Inf
+            end
+            nc += 1
+            nexpand += 1 
+            
+            u_r =  u + (1 - r) * axis # right bound
+            if all(u -> -0.5 < u < 1.5, u_r)   ## ??check if this non-periodic boundary condition is correct, it allows to exceed the unit cube 
+                v_r = prior_transform(u_r)
+                logl_r = loglike(v_r)
+            else
+                logl_r = -Inf
+            end    
```

Comment:
Same as above comment- 
```suggestion
            if all(u -> 0 < u < 1, u_r)
                v_r = prior_transform(u_r)
                logl_r = loglike(v_r)
            else
                logl_r = -Inf
            end    
```